test/TransactionInclusionProver.t.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "./BaseTest.t.sol";
+
+// libs
+import "../src/lib/RLPWriter.sol";
+import "../src/lib/RLPEncoder.sol";
+import "../src/lib/RLPReader.sol";
+import "../src/lib/Utils.sol";
+import "./utils/Data.sol";
+
+// contracts
+import "../src/interfaces/IBlockhashStorage.sol";
+import "../src/TrustedOracle.sol";
+import "../src/TransactionInclusionProver.sol";
+
+// structs
+import "../src/structs/BlockData.sol";
+import "../src/structs/ProverDto.sol";
+import "../src/structs/TxReceipt.sol";
+import "../src/structs/TxLog.sol";
+
+contract TransactionInclusionProverTest is BaseTest {
+    IBlockhashStorage public blockhashStorage;
+    TransactionInclusionProver public prover;
+
+    using RLPReader for bytes;
+    using RLPReader for uint256;
+    using RLPReader for RLPReader.RLPItem;
+    using RLPReader for RLPReader.Iterator;
+
+    function setUp() public override {
+        super.setUp();
+        blockhashStorage = new TrustedOracle();
+        prover = new TransactionInclusionProver(address(blockhashStorage));
+    }
+
+    // Tes correct scenarios
+    function testVerifyTrieProof1() public {
```

Comment:
I think we should have better names for these tests - it is good idea to say what specific unit you are testing in here.